### DISCUSSION TITLE
OpenROAD mismatch version while running with conda env

### DISCUSSION THREAD

**Discussion by vijayank88** [siliconcompiler/siliconcompiler#1180] [Help (Q&A)]
I'm trying to setup siliconcompiler through conda environment based on steps from here: https://github.com/siliconcompiler/siliconcompiler/tree/main/setup/conda
But getting error for OpenROAD mismatch version. 
```
| INFO    | job0  | syn        | 0  | Finished task in 2.43s
| INFO    | job0  | floorplan  | 0  | Tool 'openroad' found with version 'b848aaed251d09d5c855c1dbce9f189b52862707' in directory '/home/vijayan/anaconda3/envs/sc_env/bin'
| ERROR   | job0  | floorplan  | 0  | Version check failed for openroad. Check installation.
| ERROR   | job0  | floorplan  | 0  | Found version b848aaed251d09d5c855c1dbce9f189b52862707, did not satisfy any version specifier set >=v2.0-3394.
```
How to fix the above issue?

**Comment by mithro** [comment]
@proppy - Any idea?

**Comment by proppy** [comment]
Looks like silicon compiler is looking to enforce the following version check on openroad:
https://github.com/siliconcompiler/siliconcompiler/blob/654456b77bf84035e8fe29047f37753b6517c87a/siliconcompiler/tools/openroad/openroad.py#L67

Looks like you can disable this behavior with:
```
chip.set('option', 'novercheck', True)
```

**Reply by gadfort** [reply]
Ideally you shouldn't have to disable the version checking. Is it possible that the conda openroad is getting a non-standard version?

**Reply by proppy** [reply]
we build most (if not all) of the intermediate version of the all those package daily, see:
- https://anaconda.org/LiteX-Hub/openroad/files
- https://anaconda.org/LiteX-Hub/open_pdks.sky130a/files
- https://anaconda.org/LiteX-Hub/yosys/files
- https://anaconda.org/LiteX-Hub/openlane/files
- https://anaconda.org/LiteX-Hub/klayout/files
- https://anaconda.org/LiteX-Hub/magic/files
- https://anaconda.org/LiteX-Hub/netgen/files

So it should be possible to pin the corresponding blessed version in an environment file as describe here: https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-file-manually

**Reply by gadfort** [reply]
I'll try to take a look at these. I don't think we would want to pin the openroad version (without some really good reason). But it seems the version encoded into openroad (in these builds) doesn't match the normal versioning encoding and that is breaking the version check. So that might be something to explore.

**Reply by proppy** [reply]
> it seems the version encoded into openroad (in these builds)

currently the version string is generated this way:
`{{ GIT_DESCRIBE_TAG }}_{{ GIT_DESCRIBE_NUMBER }}_g{{ GIT_DESCRIBE_HASH }}-{{ DATETIME }}`

the `TAG` part of the release (which is the most meaningful bits) relies on the last public tag on the public repo (`2.0` from ~2 years ago: https://github.com/The-OpenROAD-Project/OpenROAD/tags)

I'd be great to get more recent tags published: https://github.com/The-OpenROAD-Project/OpenROAD/issues/2153

**Reply by gadfort** [reply]
I think we can update the checker to support that string format so you don't have to mess around with the `novercheck` flag (atleast for OpenROAD)

**Reply by gadfort** [reply]
@proppy could you provide me with the output of `openroad -version` from the most recent version up there? I was unable to get conda to install it on my machine due to some python version differences.

**Reply by gadfort** [reply]
This appears to be an issue with the OpenROAD build. Please compile with a modern version of git so OpenROAD gets the proper version numbering.

**Comment by vijayank88** [comment]
If i use `conda install -c LiteX-Hub openroad` It's choosing by default below version only.

` openroad           LiteX-Hub/linux-64::openroad-2.0_3654_gb848aaed2-20220512_085338`
How to force to install latest version @proppy ? Thanks

**Reply by proppy** [reply]
I suspect you might have a conflicting dependency that's preventing conda to resolve to a newer version of the package, can you file an issue against https://github.com/hdl/conda-eda so we can troubleshoot there?
